### PR TITLE
Links to buttons

### PR DESCRIPTION
**Pull request by SharkyKZ**
Pull Request for Issue # .

### Summary of Changes
This replaces link tags that look and act like buttons with actual buttons. See #20682 for details.

### Testing Instructions
Code review.

### CODE REVIEW

File: administrator/components/com_templates/helpers/html/templates.php [hunk 1]
```diff
@@ -44,8 +44,8 @@ public static function thumb($template, $clientId = 0)
 
 			if (file_exists($preview))
 			{
-				$html = '<a href="#' . $template . '-Modal" role="button" class="thumbnail pull-left hasTooltip" data-toggle="modal" title="' .
-					JHtml::_('tooltipText', 'COM_TEMPLATES_CLICK_TO_ENLARGE') . '">' . $html . '</a>';
+				$html = '<button type="button" data-target="#' . $template . '-Modal" class="thumbnail pull-left hasTooltip" data-toggle="modal"'
+					. 'title="' . JHtml::_('tooltipText', 'COM_TEMPLATES_CLICK_TO_ENLARGE') . '">' . $html . '</button>';
```
Comment (Quy) [inline administrator/components/com_templates/helpers/html/templates.php]:
```suggestion
					. ' title="' . JHtml::_('tooltipText', 'COM_TEMPLATES_CLICK_TO_ENLARGE') . '">' . $html . '</button>';
```

File: administrator/templates/hathor/html/com_templates/template/default.php [hunk 2]
```diff
@@ -176,7 +178,7 @@ function clearCoords()
 			<div class="modal-footer">
 				<form id="deleteFolder" method="post" action="<?php echo JRoute::_('index.php?option=com_templates&task=template.deleteFolder&id=' . $input->getInt('id') . '&file=' . $this->file); ?>">
 					<fieldset>
-						<a href="#" class="btn" data-dismiss="modal"><?php echo JText::_('COM_TEMPLATES_TEMPLATE_CLOSE'); ?></a>
+						<button type="button" class="btn" data-dismiss="modal"><?php echo JText::_('COM_TEMPLATES_TEMPLATE_CLOSE'); ?></button>
 						<input type="hidden" class="address" name="address" />
 						<input type="submit" value="<?php echo JText::_('COM_TEMPLATES_BUTTON_DELETE');?>" class="btn btn-danger" />
```
Comment (Quy) [inline administrator/templates/hathor/html/com_templates/template/default.php:183]: Change to `button` type?

Comment (SharkyKZ) [reply administrator/templates/hathor/html/com_templates/template/default.php:183]: Maybe in a separate PR.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 92ac1f2354839362de059bbfbf7074d565daccb7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23274">issues.joomla.org/tracker/joomla-cms/23274</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 92ac1f2354839362de059bbfbf7074d565daccb7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23274">issues.joomla.org/tracker/joomla-cms/23274</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23274">issues.joomla.org/tracker/joomla-cms/23274</a>.</sub>

**Pull request merged**